homeassistant/components/zwave_js/binary_sensor.py

```diff
@@ -0,0 +1,354 @@
+"""Representation of Z-Wave binary sensors."""
+
+import logging
+from typing import Callable, List, Optional, TypedDict
+
+from zwave_js_server.client import Client as ZwaveClient
+from zwave_js_server.const import CommandClass
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_HEAT,
+    DEVICE_CLASS_LOCK,
+    DEVICE_CLASS_MOISTURE,
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_PROBLEM,
+    DEVICE_CLASS_SAFETY,
+    DEVICE_CLASS_SMOKE,
+    DEVICE_CLASS_SOUND,
+    DOMAIN as BINARY_SENSOR_DOMAIN,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DATA_CLIENT, DATA_UNSUBSCRIBE, DOMAIN
+from .discovery import ZwaveDiscoveryInfo
+from .entity import ZWaveBaseEntity
+
+LOGGER = logging.getLogger(__name__)
+
+
+NOTIFICATION_SMOKE_ALARM = 1
+NOTIFICATION_CARBON_MONOOXIDE = 2
+NOTIFICATION_CARBON_DIOXIDE = 3
+NOTIFICATION_HEAT = 4
+NOTIFICATION_WATER = 5
+NOTIFICATION_ACCESS_CONTROL = 6
+NOTIFICATION_HOME_SECURITY = 7
+NOTIFICATION_POWER_MANAGEMENT = 8
+NOTIFICATION_SYSTEM = 9
+NOTIFICATION_EMERGENCY = 10
+NOTIFICATION_CLOCK = 11
+NOTIFICATION_APPLIANCE = 12
+NOTIFICATION_HOME_HEALTH = 13
+NOTIFICATION_SIREN = 14
+NOTIFICATION_WATER_VALVE = 15
+NOTIFICATION_WEATHER = 16
+NOTIFICATION_IRRIGATION = 17
+NOTIFICATION_GAS = 18
```

Comment:
Should these be defined in the library instead of the integration?

Reply:
Not sure, these are only the values I used to create the mappings, not a full/complete list.
It doesn't hurt though to create an enum in the lib.

Reply:
I think we can merge this as-is and if we get a more complete list move them to the lib